Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,23 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+def truncate_text_to_approximate_size(s, max_bytes, opts = {})
+  front_weight = opts[:front_weight] || 0.5
+  glue = opts[:gloe] || "\n[...snip...]\n"
+  return s if s.bytesize <= max_bytes
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
+  scale = 0.99 # just a tad small to allow for the glue
+  while true
+    front = s[0..((scale * max_bytes * front_weight).floor)]
+    back_start = s.size-((scale * max_bytes * (1.0-front_weight)).ceil)
+    back = s[back_start..s.size]
+    out = front + glue + back
+    return out if out.bytesize <= max_bytes
```

Comment:
I'd probably flip this around to do the `raise` here instead so `out` is the final statement and it's successful.